client/src/components/Messaging/NewMessageForm.js

```diff
@@ -30,7 +32,7 @@ export const NewMessageForm = ({
       className={`${styles.sendMessageForm} ${!darkMode && styles.lightsendMessageForm}`}
     >
       <div className={styles.emojiMessageContainer}>
-        {!detectMobile() && <EmojiRow text={text} setText={setText} />}
+        {emojiRow && !detectMobile() && <EmojiRow text={text} setText={setText} />}
```

Comment:
This will cause a layout shift which will lead to bad flicker. I would suggest adding a container with a fixed height so that the layout doesn't change when changing visibility.

Reply:
i tried to implement what u suggested , can you take a look ?  :)